dlc/src/main/scala/org/bitcoins/dlc/DLCMessage.scala

```diff
@@ -2,28 +2,234 @@ package org.bitcoins.dlc
 
 import org.bitcoins.core.crypto.{ECPublicKey, ExtPublicKey, Sha256DigestBE}
 import org.bitcoins.core.currency.Satoshis
-import org.bitcoins.core.protocol.Bech32Address
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.BlockStamp.BlockTime
 import org.bitcoins.core.protocol.transaction.{
   TransactionOutPoint,
   TransactionOutput
 }
+import org.bitcoins.core.protocol.{Bech32Address, BlockStamp, NetworkElement}
+import org.bitcoins.core.psbt.InputPSBTRecord
+import org.bitcoins.core.psbt.InputPSBTRecord.PartialSignature
+import org.bitcoins.core.util.{CryptoUtil, Factory}
 import org.bitcoins.core.wallet.fee.SatoshisPerVirtualByte
+import scodec.bits.ByteVector
+import ujson._
+
+import scala.annotation.tailrec
+import scala.collection.mutable
 
 sealed trait DLCMessage
 
 object DLCMessage {
+
+  // TODO: will need to be changed when this is standardized
+  def calcEventId(
+      oracleInfo: OracleInfo,
+      contractInfo: ContractInfo,
+      timeouts: DLCTimeouts): String = {
+    CryptoUtil
+      .sha256(oracleInfo.bytes ++ contractInfo.bytes ++ timeouts.bytes)
+      .hex
+  }
+
   case class OracleInfo(pubKey: ECPublicKey, rValue: ECPublicKey)
+      extends NetworkElement {
+
+    override def bytes: ByteVector = pubKey.bytes ++ rValue.bytes
+  }
+
+  object OracleInfo extends Factory[OracleInfo] {
+
+    override def fromBytes(bytes: ByteVector): OracleInfo = {
+      require(bytes.size == 66, s"OracleInfo is only 66 bytes, got $bytes")
+
+      val pubkey = ECPublicKey(bytes.take(33))
+      val rValue = ECPublicKey(bytes.drop(33))
+
+      // TODO: validate rValue once the libsecp functionality is added
+
+      OracleInfo(pubkey, rValue)
+    }
+  }
+
+  case class ContractInfo(contractInfo: Map[Sha256DigestBE, Satoshis])
```

Comment:
`contractInfo.contractInfo` is an anti-pattern so you should rename the inner thing to `outcomeValueMap` or something like this and also consider giving the case class the common map methods so that users don't have to reach in every time